ibis_substrait/compiler/translate.py

```diff
@@ -777,13 +777,27 @@ def selection(
 
     if selections := _get_selections(op):
 
-        if relation.project.common.ListFields():
-            # if there is already an `emit` in RelCommon then we're stacking
-            # projections and we need to update the output_mapping to refer to
-            # the number of fields present in the most recent emit
-            mapping_counter = itertools.count(
-                len(relation.project.common.emit.output_mapping)
+        rels = [
+            getattr(relation, attr)
+            for attr in (
+                "aggregate",
+                "cross",
+                "extension_leaf",
+                "extension_multi",
+                "extension_single",
+                "fetch",
+                "filter",
+                "join",
+                "project",
+                "read",
+                "set",
+                "sort",
```

Comment:
I would be more clever about this, but there's already enough "cleverness" in `protobuf` that I opted for simple.